### PR TITLE
[SPARK-44052][CONNECT][PS] Add util to get proper Column or DataFrame class for Spark Connect.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to add util to get proper Column or DataFrame class for Spark Connect.

### Why are the changes needed?

To eliminate code duplication and obtain the appropriate Spark Connect Column and Spark Connect DataFrame classes, utility functions `get_column_class` and `get_dataframe_class` have been added.

### Does this PR introduce _any_ user-facing change?

No, it's code refactoring.

### How was this patch tested?

The existing CI should pass.

### CODE REVIEW

File: python/pyspark/pandas/indexes/multi.py [hunk 1]
```diff
@@ -545,14 +555,23 @@ def _is_monotonic_increasing(self) -> Series:
...
         window = Window.orderBy(NATURAL_ORDER_COLUMN_NAME).rowsBetween(-1, -1)
 
         cond = F.lit(True)
         has_not_null = F.lit(True)
+        if is_remote():
+            from pyspark.sql.connect.column import Column as ConnectColumn
+
+            Column = ConnectColumn
+        else:
+            Column = PySparkColumn  # type: ignore[assignment]
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/indexes/multi.py:574]: Shall we also make a utility function to get the attribute from `Column`?

Comment (itholic) [reply python/pyspark/pandas/indexes/multi.py]: Just added `get_column_class` and `get_dataframe_class`, applied to whole code path. Also updated the PR title and description. Thanks!

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/pandas/tests/connect/indexes/test_parity_base.py [hunk 2]
```diff
@@ -35,14 +35,16 @@ def psdf(self):
     def test_append(self):
         super().test_append()
 
+    @unittest.skip(
+        "TODO(SPARK-43611): Fix unexpected `AnalysisException` from Spark Connect client."
+    )
```
Comment (itholic) [inline python/pyspark/pandas/tests/connect/indexes/test_parity_base.py:40]: Well... it's wired that the test failure on Github Action is not reproducible from my local, so I merge this into other ticket instead.

Updated the PR title & PR description, and will address this in separate ticket with other tickets that have same cause.

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**